bin/io.js

```diff
@@ -0,0 +1,56 @@
+// this module just contains stuff used for io
+const fs = require('fs');
+const util = require('util');
+const httpMessageParser = require('http-message-parser');
+
+/**
+ * This opens a file from disk asynchronously.
+ *
+ * @param {string} path - A file path.
+ *
+ * @returns {string|Buffer} The result of opening the file.
+ */
+const readFile = util.promisify(fs.readFile);
+exports.readFile = readFile;
+
+/**
+ * Simple wrapper around node's process.stdin.
+ * This allows us to get the result of cat and other commands.
+ *
+ * @param {string} [encoding='utf8'] - The http message's encoding.
+ *
+ * @returns {Promise<string>} Should return the http message.
+ */
+async function getStdin(encoding = 'utf8') {
+  const {stdin} = process;
+  let message = '';
+  return new Promise((resolve, reject) => {
+    try {
+      stdin.setEncoding(encoding);
+      stdin.on('readable', () => {
+        let chunk;
+        while((chunk = stdin.read())) {
+          message += chunk;
+        }
+      });
+      stdin.on('end', () => resolve(message));
+      stdin.on('error', reject);
+    }
+    catch(e) {
+      reject(e);
+    }
+  });
+}
+exports.getStdIn = getStdin;
+
+async function getHTTPMessage() {
+  const HTTPMessage = await getStdin();
+  if(!HTTPMessage) {
```

Comment:
maybe `getStdin` should throw on no input instead?

Reply:
no, stdin throws right now if the file is not there.
http message throws if there is no http message.
I think this is better because we know stdin succeeded it's just you forgot
to send in an http message.